### ISSUE TITLE
Closing the door after the client left?

### ISSUE THREAD

**Issue by bensoy** [jvinet/knock#20]
Instead of using another combination, how can we close the door and reset the chain after client disconnects from our secure port? I couldn't see something like that in docs, it would be great feature if its not already :)

**Comment by jshall** [comment]
I tried something like this...
```
[openSSH]
	sequence    = 4444:udp,3333:udp,22
	seq_timeout = 15
	command     = iptables -t nat -A WANPREROUTING -p tcp --dport 22 -j DNAT --to-destination 192.168.0.2; iptables -A wanin -s %IP% -p tcp -d 192.168.0.2 --dport 22 -j ACCEPT

[closeSSH]
	sequence    = 22
	tcpflags    = fin
	command     = iptables -t nat -D WANPREROUTING -p tcp --dport 22 -j DNAT --to-destination 192.168.0.2; iptables -D wanin -s %IP% -p tcp -d 192.168.0.2 --dport 22 -j ACCEPT
```
+ `[openSSH]` works just fine.
+ `[closeSSH]` recognizes Stage 1 at the appropriate time,
 but then stalls out like it can't believe there is not a Stage 2.

**Comment by barnardm** [comment]
Here is how I have solved this problem on my servers.
1) Setup IPTables to allow established connections to stay on
```sudo iptables -A INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT```
2) Tell knockd to disconnect after a sec period of time
```
[SSH]
        sequence    = 4444:udp,3333:udp,22
        tcpflags    = syn
        seq_timeout = 15
        start_command = /sbin/iptables -I INPUT 1 -s %IP% -p tcp --dport 22 -j ACCEPT
        cmd_timeout = 10
        stop_command = /sbin/iptables -D INPUT -s %IP% -p tcp --dport 22 -j ACCEPT
```

With this setup you do not need a start/stop sequence. cmd_timeout is 10 seconds, could be adjusted +/- for convenience. 

How it works:
* provide the knock sequence
* 10 seconds later your start command is removed from iptables
* Your connection will stay active because of the established/related rule added to iptables.

This doesn't wait for the connection to close but basically solves the same problem.

Found this method in a digital ocean tutorial:
https://www.digitalocean.com/community/tutorials/how-to-use-port-knocking-to-hide-your-ssh-daemon-from-attackers-on-ubuntu
